packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -888,8 +904,30 @@ private void destroyMapViewIfNecessary() {
     if (mapView == null) {
       return;
     }
-    mapView.onDestroy();
+
+    final MapView mapReference = mapView; // keep a reference to the mapView for the callback
     mapView = null;
+
+    // This fixes an issue where the mapView is still being used by the render thread after disposal.
+    // Delaying the actual mapView disposal to the next frame avoids the issue.
+    Runnable r =
+        () -> {
+          Log.e(TAG, "Mapview destroy in callback");
+          mapReference.onDestroy();
+        };
+
+    OnMapsSdkInitializedCallback mapCallback =
+        renderer -> {
+          if (renderer == MapsInitializer.Renderer.LATEST) {
+            Log.e(TAG, "Renderer LATEST");
+            handler.post(r);
+          } else if (renderer == MapsInitializer.Renderer.LEGACY) {
+            Log.e(TAG, "Renderer LEGACY");
+            handler.postDelayed(r, 100);
```

Comment:
100 seems very short, especially since according to comments in the issue it seems like there's a relationship here to animations running, which could extend things somewhat arbitrarily. Something longer, like a full second of delay, might be safer.